content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,123 @@
+---
+title: 'Prisma Client extensions'
+metaTitle: 'Prisma Client extensions (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+preview: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+Use Prisma Client extensions to add functionality to your models, result objects, queries, or to Prisma Client itself.
+
+There are four types of extension. You can use each type to customize a different aspect of Prisma Client.
+
+- `model` extensions: [add custom methods or fields to your models](/concepts/components/prisma-client/client-extensions/model)
+- `client` extensions: [add client-level methods to Prisma Client](/concepts/components/prisma-client/client-extensions/client)
+- `result` extensions (for future release): [add custom results objects to Prisma Client](/concepts/components/prisma-client/client-extensions/result)
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To create Prisma Client extensions, use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods), and the appropriate extension type (`model`, `client`, `result` or `query`).
+
+For example, the following code uses the `model` extension type to extend the `user` model:
+
+```ts
+const xprisma = prisma.$extends({
+  name: `signUp`,  // (Optional) This extension name appears in error logs and tracing
+  model: {  // This is a model type extension
+    user: { ... } // The extension logic goes inside the curly braces
+ },
+})
+```
+
+`name` is an optional field that you can use to name the extension. This name appears in error logs and tracing.
+
+You can use [multiple independent extensions at the same time](#multiple-extensions). Each of these instances runs in is own isolated instance of Prisma Client, unless you deliberately combine or stack them.
+
+You can also [share your Prisma Client extensions](/concepts/components/prisma-client/client-extensions/shared-extensions) with other Prisma users, and [import Prisma Client extensions developed by other users](#install-a-packaged-extension) into your Prisma project.
+
+## Prerequisites
+
+To use Prisma Client extensions, you must enable the `clientExtensions` preview feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Multiple extensions
+
+Each time that you call `$extends`, you create an _extended client_. This extended client is a forked variant of the standard Prisma Client. The original client does not change.
+
+Every extended client operates independently, in an isolated instance. It cannot conflict with other extended clients or the original client.
```

Comment:
This explanation is great - but for all users of extensions, not only for the ones that want multiple extensions. I think it should be moved to the section above.

(This might also be a good place that they all talk to the same Prisma Query Engine, and do not get their own. So only 1 connection pool for all forked clients.)

Reply:
Great minds think alike... I've just moved this information to the general intro